### PR TITLE
Make beam_constraints.Constraint.advance() docstring more accurate

### LINKED ISSUE #32668: Constraint.advance() docstring is slighly incorrect
### System Info

according to the implementations of the Constraint ABC, the [advance() ](https://github.com/huggingface/transformers/blob/c1357834e86e71792d1cad06a998f5ebf6e07126/src/transformers/generation/beam_constraints.py#L49)function can return either a single token id (as the docstring indicates) or a list of token ids that would advance the beam search. It should be updated to reflect this.

Here is an example fix:

```
@abstractmethod
def advance(self):
    """
    When called, returns the token(s) that would take this constraint one step closer to being fulfilled.

    Return:
        token_ids (Union[int, List[int], None]): 
            - A single token ID (int) that advances the constraint
            - A list of token IDs that could advance the constraint
            - None if the constraint is completed or cannot be advanced
    """
```

### Who can help?

_No response_

### Information

- [ ] The official example scripts
- [ ] My own modified scripts

### Tasks

- [ ] An officially supported task in the `examples` folder (such as GLUE/SQuAD, ...)
- [ ] My own task or dataset (give details below)

### Reproduction

Just a docstring change.

### Expected behavior

Should be updated for accuracy, easy of implementing this abstract base class.

### PR DESCRIPTION
**Pull request by alex-calderwood**
# What does this PR do?
The abstract base class Constraint.advance() docstring indicated that it could only return an integer token_id, when in fact it can return a list of token_ids.

Fixes #32668
https://github.com/huggingface/transformers/issues/32668

## Before submitting
- [x] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [ ] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/main/CONTRIBUTING.md#create-a-pull-request),
      Pull Request section?
- [ ] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link
      to it if that's the case.
- [ ] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/transformers/tree/main/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/main/docs#writing-source-documentation).
- [ ] Did you write any new necessary tests?

## Who can review?

Anyone in the community is free to review the PR once the tests have passed. Feel free to tag
members/contributors who may be interested in your PR.

@gante @stevhliu

### CODE REVIEW

File: src/transformers/generation/beam_constraints.py [hunk 1]
```diff
@@ -48,10 +48,13 @@ def test(self):
     @abstractmethod
     def advance(self):
         """
-        When called, returns the token that would take this constraint one step closer to being fulfilled.
+        When called, returns the token(s) that would take this constraint one step closer to being fulfilled.
 
         Return:
-            token_ids(`torch.tensor`): Must be a tensor of a list of indexable tokens, not some integer.
+            token_ids (Union[int, List[int], None]):
```
Comment (stevhliu) [inline src/transformers/generation/beam_constraints.py:54]: Trailing whitespace is causing the doc-builder to fail

```suggestion
            token_ids (Union[int, List[int], None]):
```

**Review by stevhliu — APPROVED** [review]
Thanks for the fix! We can merge once @gante approves as well 🙂

**Review by gante — APPROVED** [review]
Yup, that's correct, thank you for fixing 🤗
